In [1]:
import pandas as pd
import numpy as mp
import scipy.stats as stats

### Q1

A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

Null Hypothesis : 𝜇1 = 𝜇2

Alternative Hypothesis : 𝜇1 ≠ 𝜇2



In [3]:
Cutlets = pd.read_csv('Cutlets.csv')
Cutlets.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [4]:
stats.ttest_ind(Cutlets['Unit A'],Cutlets['Unit B'],alternative = 'two-sided')

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

pvalue is greater than significance level ,that is 0.4722 > 0.05

#### Therefore , there is no significant in the diameter of the cutlet between two units.

## Q2

A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
 


In [16]:
LabTAT = pd.read_csv ('LabTAT.csv')
LabTAT.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [17]:
stats.f_oneway(LabTAT.iloc[:,0], LabTAT.iloc[:,1], LabTAT.iloc[:,2], LabTAT.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

### Q3

Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across region

Null hypothesis        : All proportions are equal

Alternative Hypothesis : not all proportions are equal

In [22]:
BuyRatio = pd.read_csv('BuyerRatio.csv')
BuyRatio

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [36]:
BuyR_new = BuyRatio.iloc[0:,1:5]

In [37]:
stats.chi2_contingency(BuyR_new)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

p_value = 0.66
Alpha   = 0.05

Therefore, p_value > Alpha,
So we accept Null hypothesis

#### Hence we can conclude that all proportions are equal

### Q4

TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences.

Null Hypothesis        : The defective % varies by centre.

Alternative Hypothesis : The defective % does not varies by centre.


In [31]:
Cstm_Form =  pd.read_csv('Costomer+OrderForm.csv')
Cstm_Form.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [48]:
print(Cstm_Form['Phillippines'].value_counts(),
Cstm_Form['Indonesia'].value_counts(),
Cstm_Form['Malta'].value_counts(),
Cstm_Form['India'].value_counts())

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 Error Free    269
Defective      31
Name: Malta, dtype: int64 Error Free    280
Defective      20
Name: India, dtype: int64


In [52]:
observed_Value = [[271,267,269,280],[29,33,31,20]]

In [54]:
stats.chi2_contingency(observed_Value)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

p_value = 0.2771

Alpha   = 0.05

p_value is greater than alpha.There for we accept Null Hypothesis

#### Hence we can conciude that the defective varies by centre.